### PR TITLE
x:ClassModifier support for XamlLoader

### LINKED ISSUE #8971: Avalonia's GenerateResourcesTask shows on Minimal verbosity
**Describe the bug**
In AvaloniaBuildTasks.target this line
image
sets the MessageImportance to low if it's not set - which by MSBuild's documentation should make it invisible in the Minimal log verbosity.

However, the only way to not show this message is to set the log verbosity to Quiet which is undesirable for my use case.

**To Reproduce**
Build any avalonia project (for example Control Catalog) with verbosity set to Minimal. Observe the "avares-> ..." lines in the build log.

**Expected behavior**
Since the priority is low, these messages should only be visible in high verbosity logs.

**Screenshots**
image

**Additional context**
In a project with many resources that need to be generated (mine has over 200 different .axaml files) this creates a lot of clutter and becomes very inconvenient for users.

### LINKED ISSUE #11516: Remove generated XamlLoader from Simple and Fluent themes
**Is your feature request related to a problem? Please describe.**

For context, it breaks StyleInclude and ResourceInclude to the internal files of the themes. Like `<ResourceInclude Source='avares://Avalonia.Themes.Fluent/FluentControls/Button.xaml' />`.

We have many ideas on default themes improvement that we are not able to do before 11.0 release. Mostly it includes changes to what exactly is generated by the XAML compiler, making controls trimmable, improving theme assembly size, improving startup performance, etc... Some of these changes require changing the internal structure and how dictionaries are organized.

Removing XamlLoader from default themes is a breaking change that might be annoying for some, but should be acceptable in overall. 

In 0.10.x timeline we have not considered the internals of the themes as a public API, even though it was possible to use StyleInclude/ResourceInclude to them. **In WPF/UWP theme internals were never publicly available at all**. 

It's still possible to use only parts of the theme, by creating FluentTheme/SimpleTheme objects and accessing their content with TryFindResource instead of assigning the whole theme to the Application. This won't be changed.

Previously, there was a use case to use only a part of the Fluent theme in order to override some resources. For instance, include all files except AccentColors.xaml, which would be replaced with a custom one. Now it's not needed, as developers can override ColorPaletteResources for the fluent theme providing custom accent color or base colors (todo - add documentation link here).

Note, it doesn't affect their party controls and in-app XAML files by default.

### PR DESCRIPTION
**Pull request by maxkatz6**
## What is the current behavior?

All XAML files are public and available everywhere.
Even if you declare x:Class with an internal modifier on C# side - it still can be accessible via XamlLoader (StyleInclude and ResourceInclude).

## What is the updated/expected behavior with this PR?

1. x:ClassModifier is parsed in the XAML compiler as a supported directive.
2. Not necessary, but added a validation if anybody uses both x:Class type c# modifier and x:ClassModifier with not matching values.
3. If XAML is not public (either x:ClassModifier or x:Class) it won't produce XamlLoader entries.
4. If XAML is not public, generated NamespaceInfo, Build and Populate methods are internal.
5. Updated all Simple and Fluent XAML files to be x:ClassModifier="internal" except entry point classes - FluentTheme and SimpleTheme which are supposed to be a public API.

6. Note, x:ClassModifier DOES NOT affect Xaml Name Reference Generator, as this generator requires x:Class and follows x:Class modifiers. Meaning, x:ClassModifier is useless there - just change modifier on C# type.

## Breaking changes

Yes. StyleInclude and ResourceInclude do not work if a developer wants to access now internal files inside of these two assemblies.

## Fixed issues

Closes #11516
Closes #8971 as a bonus

### CODE REVIEW

File: src/Avalonia.Build.Tasks/GenerateAvaloniaResourcesTask.cs [hunk 1]
```diff
@@ -127,7 +127,7 @@ private bool PreProcessXamlFiles(List<Source> sources)
 
         public bool Execute()
         {
-            Enum.TryParse<MessageImportance>(ReportImportance, out _reportImportance);
+            Enum.TryParse(ReportImportance, true, out _reportImportance);
```
Comment (maxkatz6) [inline src/Avalonia.Build.Tasks/GenerateAvaloniaResourcesTask.cs:130]: Noticed this minor issue causing #8971. Decided to not open another PR for this one liner.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0035541-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by maxkatz6** [comment]
Another side bonus.
With these changes Avalonia.Themes.Fluent assembly can be trimmed further than before. As unused and already merged classes are removed on trimming now. **From 1689kb to 728kb.**

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0035545-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0035555-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: samples/Sandbox/Sandbox.csproj [hunk 2]
```diff
@@ -5,6 +5,7 @@
     <TargetFramework>net6.0</TargetFramework>
     <TargetLatestRuntimePatch>true</TargetLatestRuntimePatch>
     <IncludeAvaloniaGenerators>true</IncludeAvaloniaGenerators>
+<!--    <AvaloniaXamlIlDebuggerLaunch>true</AvaloniaXamlIlDebuggerLaunch>-->
```
Comment (kekekeks) [inline samples/Sandbox/Sandbox.csproj:8]: nit

**Review by kekekeks — APPROVED** [review]

**Comment by robloo** [comment]
> With these changes Avalonia.Themes.Fluent assembly can be trimmed further than before. As unused and already merged classes are removed on trimming now. From 1689kb to 728kb.

More than 50%, that's more than a side-bonus for web!

**Pull request merged**